bentoml/service/__init__.py

```diff
@@ -248,16 +248,24 @@ def env_decorator(
         zipimport_archives: list of zipimport archives paths relative to the module path
     """
 
-    if pip_dependencies is not None:
-        logger.warning(
-            "`pip_dependencies` parameter in `@env` is being deprecated soon, use "
-            "`pip_packages` instead, e.g. `@env(pip_packages=[\"numpy\"])`"
-        )
-    if auto_pip_dependencies is not None:
-        logger.warning(
-            "`auto_pip_dependencies` parameter in `@env` is being deprecated soon, use"
-            "`infer_pip_packages` instead, e.g. `@env(infer_pip_packages=True)`"
-        )
+    if requirements_txt_file:
+        if pip_packages:
+            logger.warning("Ignoring pip_packages as requirements_txt_file is set.")
+        if pip_index_url or pip_trusted_host or pip_extra_index_url:
+            logger.warning(
+                "Ignoring pip related options as requirements_txt_file is set."
+            )
+    else:
```

Comment:
why this `else`  here? I think those warnings are orthogonal to the `requirements_txt_file` related warnings above right?